lib/AST/Decl.cpp

```diff
@@ -5042,3 +5042,20 @@ getObjCNameForSwiftDecl(const ValueDecl *VD, DeclName PreferredName){
     return {Ctx.getIdentifier(getNameForObjC(VD)), ObjCSelector()};
   }
 }
+
+bool swift::objc_translation::
+isVisibleToObjC(const ValueDecl *VD, Accessibility minRequiredAccess,
+                 bool checkParent) {
+  if (!(VD->isObjC() || VD->getAttrs().hasAttribute<CDeclAttr>()))
+    return false;
+  if (VD->getFormalAccess() >= minRequiredAccess) {
+    return true;
+  } else if (checkParent) {
+    if (auto ctor = dyn_cast<ConstructorDecl>(VD)) {
+      // Check if we're overriding an initializer that is visible to obj-c
```

Comment:
This is the logic PrintAsObjC was using to decide whether to include a decl in the generated header – I just moved the function here to make sure USR generation and the header printer use the same logic. I assume 'checkParent' was added specifically for the case of still writing out < minRequiredAccess initializers that override a >= minRequiredAccess initializer.